### ISSUE TITLE
Timeline API: Exception: {"Requested value 'reviewed' was not found."}

### ISSUE THREAD

**Issue by aniket-aurea** [octokit/octokit.net#1525]
Note: gitHubClient.PullRequest.Get works fine for the same pull request
StackTrace:
   at System.Enum.EnumResult.SetFailure(ParseFailureKind failure, String failureMessageID, Object failureMessageFormatArgument)
   at System.Enum.TryParseEnum(Type enumType, String value, Boolean ignoreCase, EnumResult& parseResult)
   at System.Enum.Parse(Type enumType, String value, Boolean ignoreCase)
   at Octokit.Internal.SimpleJsonSerializer.GitHubSerializerStrategy.DeserializeObject(Object value, Type type) in C:\Users\shiftkey\code\octokit.net\Octokit\Http\SimpleJsonSerializer.cs:line 125
   at Octokit.PocoJsonSerializerStrategy.DeserializeObject(Object value, Type type) in C:\Users\shiftkey\code\octokit.net\Octokit\SimpleJson.cs:line 1478
   at Octokit.Internal.SimpleJsonSerializer.GitHubSerializerStrategy.DeserializeObject(Object value, Type type) in C:\Users\shiftkey\code\octokit.net\Octokit\Http\SimpleJsonSerializer.cs:line 166
   at Octokit.PocoJsonSerializerStrategy.DeserializeObject(Object value, Type type) in C:\Users\shiftkey\code\octokit.net\Octokit\SimpleJson.cs:line 1505
   at Octokit.Internal.SimpleJsonSerializer.GitHubSerializerStrategy.DeserializeObject(Object value, Type type) in C:\Users\shiftkey\code\octokit.net\Octokit\Http\SimpleJsonSerializer.cs:line 166
   at Octokit.SimpleJson.DeserializeObject(String json, Type type, IJsonSerializerStrategy jsonSerializerStrategy) in C:\Users\shiftkey\code\octokit.net\Octokit\SimpleJson.cs:line 583
   at Octokit.SimpleJson.DeserializeObject[T](String json, IJsonSerializerStrategy jsonSerializerStrategy) in C:\Users\shiftkey\code\octokit.net\Octokit\SimpleJson.cs:line 595
   at Octokit.Internal.SimpleJsonSerializer.Deserialize[T](String json) in C:\Users\shiftkey\code\octokit.net\Octokit\Http\SimpleJsonSerializer.cs:line 21
   at Octokit.Internal.JsonHttpPipeline.DeserializeResponse[T](IResponse response) in C:\Users\shiftkey\code\octokit.net\Octokit\Http\JsonHttpPipeline.cs:line 62
   at Octokit.Connection.<Run>d__53`1.MoveNext() in C:\Users\shiftkey\code\octokit.net\Octokit\Http\Connection.cs:line 560
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at Octokit.ApiConnection.<GetPage>d__40`1.MoveNext() in C:\Users\shiftkey\code\octokit.net\Octokit\Http\ApiConnection.cs:line 0
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at Octokit.ApiConnection.<>c__DisplayClass17_0`1.<<GetAll>b__0>d.MoveNext() in C:\Users\shiftkey\code\octokit.net\Octokit\Http\ApiConnection.cs:line 0
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at Octokit.ApiPagination.<GetAllPages>d__0`1.MoveNext() in C:\Users\shiftkey\code\octokit.net\Octokit\Clients\ApiPagination.cs:line 0

**Comment by shiftkey** [comment]
@aniket-aurea do you have an example URL that I can use to confirm this is related to [`EventInfoState`](https://github.com/octokit/octokit.net/blob/c9b2c1260bc87b7782d4fd9645d43a8885203923/Octokit/Models/Response/EventInfo.cs#L74) not having this value?

**Comment by ryangribble** [comment]
Apart from adding enum values that are missing, we also have a proposal to handle these at runtime here  #1504

**Comment by aniket-aurea** [comment]
Thanks, this sounds great.

I don't have a url as this is a private repo.
But I saw atleast 1 other status apart from the one mentioned in the
ticket, that gave the same exception.

Yes the problem did occur during de-serialization by Newtonsoft, as the
Enum did not have the value.
I have made this field string in my classes till a fix is available, and am
using the rest API till then.

Hope this helps.






[image: Aurea] <http://go.aurea.com/e/23432/2015-11-05/3brlh6/1191512291>

*Aniket Ramekar* | .Net Architect
aniket.ramekar@aurea.com | +91 9819928411
*Skype:* jilted4eva

www.aurea.com
